lib/cartopy/mpl/geoaxes.py

```diff
@@ -642,7 +641,7 @@ def set_extent(self, extents, crs=None):
         # extents are obviously the same as the projection domain.
         try_workaround = ((crs is None and
                            isinstance(self.projection, ccrs.PlateCarree)) or
-                          crs == self.projection)
+                          isinstance(crs, ccrs.PlateCarree))
```

Comment:
This doesn't check the same thing does it? It was supposed to be checking if the given crs is the same as the projection (i.e. native coordinates, don't care what the projection actually is). Now it is just checking if the crs is `PlateCarree`, which seems wouldn't be useful unless we also knew the projection was `PlateCarree`. Can you explain the motivation here?

Reply:
@ajdawson the motivation is that according to @corinnebosley the original is not working...

Reply:
That doesn't help at all. What I want to know is why are we changing the logic (i.e. the question we are asking) instead of the implementation of the logic?

It seems like the question we want the answer to is "is the specified crs the same as our projection?". Instead this change checks if the crs is `PlateCarree`, but why? How does this help determine if the workaround is needed? What if the specified crs matches the projection but isn't `PlateCarree`?

Reply:
Yes thank you @ajdawson, Pete already pointed out that observation.  It was an oversight on my part which I am now fixing.  I do still want to test that the crs is the same as the projection no matter what they are.

Reply:
OK, timing difference between starting comment and posting, PR had 0 comments when I began.